backend/src/apiserver/resource/resource_manager.go

```diff
@@ -286,6 +286,8 @@ func (r *ResourceManager) CreateRun(apiRun *api.Run) (*model.RunDetail, error) {
 
 	// Disable istio sidecar injection
 	workflow.SetAnnotationsToAllTemplates(util.AnnotationKeyIstioSidecarInject, util.AnnotationValueIstioSidecarInjectDisabled)
+	// Add a KFP specific annotation for cache service filtering
+	workflow.SetAnnotationsToAllTemplates(util.AnnotationKeyCacheEnabled, util.AnnotationValueCacheEnabled)
```

Comment:
If we always set it to true, users won't be able to  override step by step.
Is this what you intended?

Reply:
Okay I think here the naming is a bit of confusing. So the purpose of this annotation is to do pod filtering instead of actually determine the cache behavior. User can still set the MaxCacheStaleness step by step.

Reply:
This annotation here is to indicate whether this run will go through cache service. We have another two checkpoint for step cache itself: 1) template hash to determine whether this step is the same as previous one 2)MaxCacheStaleness to determine with cache entry to use.

Reply:
I see, so `MaxCacheStaleness` will be used to disable steps conditionally.
Thanks for confirming this!
sgtm

Reply:
So maybe rename to `CacheQualified`?

Reply:
I didn't know MaxCacheStaleness so I got the confusion, I think the naming is fine (and accurate too, becuase it actually means caching is enabled, just that MaxCacheStaleness = 0 will never use cache even if cache is enabled.)

Reply:
I also think that cache enabled is easier to understand.